### ISSUE TITLE
Not able to start vanilla nodejs server: Error: EBADF: bad file descriptor, write

### ISSUE THREAD

**Issue by cliffordfajardo** [stackblitz/webcontainer-core#478]
### Describe the bug

When I attempt to start my node server with the following command in the stackblitz terminal, I recieve an error:

```sh
> node index.js
[Error: EBADF: bad file descriptor, write] {
  syscall: 'write',
  errno: -9,
  code: 'EBADF'
}
```

### Link to the blitz that caused the error

https://stackblitz.com/edit/node-dzncag?file=package.json

### Steps to reproduce

1. Run `node index.js` inside the stackblitz terminal

### Expected behavior

It should run my index.js file

### Platform

```
Browser name  = Chrome 
Full version  = 96.0.4664.93
Major version = 96
navigator.appName = Netscape
navigator.userAgent = Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/96.0.4664.93 Safari/537.36
performance.memory = {
  "totalJSHeapSize": 77739238,
  "usedJSHeapSize": 72282378,
  "jsHeapSizeLimit": 4294705152
}
Hash = c1a2f33ebdb80ec1573b6337c828d176e80f0398
```

**Comment by ntucker** [comment]
In case others stumble in here because they used webpack thread-loader - removing it solved my case of `EBADF: bad file descriptor`. I'm sure there are other possible reasons. Might be helpful to get a longer stack trace or something in these cases.

**Comment by pjoe** [comment]
Got same error when trying fastify.

Disabling logging or specifying file path like `/tmp/log` works.

Also tried:
- `file: '/dev/stdout'`: fails with EBADF
- `stream: process.stdout`: fails with EBADF

suspect this may be related to how file descriptors are handled for stdout 🤔

**Comment by pjoe** [comment]
Found a workaround (probably bypassing some of fastify's optimizations):
```js
const fastify = Fastify({
  logger: {
    stream: {
      write(msg) {
        process.stdout.write(msg);
      },
    },
  },
});
```

**Comment by ntucker** [comment]
Would be great to have some basic docs on the limitations of this. It appears like maybe it's just accessing any part of the file system that isn't in `/tmp`?

**Comment by jrvidal** [comment]
> suspect this may be related to how file descriptors are handled for stdout

You're right on the money, that's the limitation you're hitting. I wanna reiterate that we know of this issue, we'll fix it as soon as we have the bandwidth for it :crossed_fingers:

**Comment by SamVerschueren** [comment]
This issue was fixed a while ago and verified that your fastify example works now 🎉 .

**Comment by AntoninLeroy** [comment]
I have the same error trying to run 'npm run build' on a vue js github connected repo.
Any ideas why ?

**Comment by jrvidal** [comment]
:wave: @AntoninLeroy Could you open a separate issue?
